pkg/controller/remotemachineset/gcpactuator.go

```diff
@@ -351,3 +380,35 @@ func (a *GCPActuator) findAvailableLeaseChars(cd *hivev1.ClusterDeployment, leas
 
 	return keys, nil
 }
+
+func requireLeases(clusterVersion string, remoteMachineSets []machineapi.MachineSet, logger log.FieldLogger) bool {
+	logger = logger.WithField("clusterVersion", clusterVersion)
+	if v, err := semver.ParseTolerant(clusterVersion); err == nil {
```

Comment:
My thinking on this was two-fold.
1) Not requiring a lease should be the default behavior going forward.
2) We know that for existing OpenShift versions, the cluster version is a valid semver. This may change in the future. But, those future versions will not be using leases. So this makes a more resilient to possible (albeit unlikely) changes to the format of the cluster version in the future.